src/WebJobs.Script.WebHost/Configuration/HostHstsOptionsSetup.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.HttpsPolicy;
+using Microsoft.Azure.WebJobs.Script.Configuration;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Configuration
+{
+    internal class HostHstsOptionsSetup : IConfigureOptions<HostHstsOptions>
+    {
+        private readonly IConfiguration _configuration;
+
+        public HostHstsOptionsSetup(IConfiguration configuration)
+        {
+            _configuration = configuration;
+        }
+
+        public void Configure(HostHstsOptions options)
+        {
+            IConfigurationSection jobHostSection = _configuration.GetSection(ConfigurationSectionNames.JobHost);
+            var hstsSection = jobHostSection.GetSection(ConfigurationSectionNames.Hsts);
+
+            if (hstsSection.Exists())
+            {
+                var hstsIsEnabled = hstsSection.GetValue<bool?>(ScriptConstants.HstsIsEnabledPropertyName);
+                if (!hstsIsEnabled.HasValue)
+                {
+                    string message = $"The value of IsEnabled property in hsts section of {ScriptConstants.HostMetadataFileName} file is missing. See https://aka.ms/functions-hostjson for more information";
+                    throw new ArgumentException(message);
+                }
```

Comment:
Do we do this anywhere else? I'm not sure we need to log if something isn't correct in config... it'll just silently no-op.

Reply:
We do this in a couple of places in `ScriptHostOptionsSetup` and `ExtensionBundleConfigurationHelper`. The reason we need to do it here is because we use `IsEnabled` to control whether we want use hsts configuration or not. So if somebody adds hsts configuration without isEnabeld configured and we silently continue they would think is hsts configured where in reality it is not. This can lead to some confusion.

- **ScriptHostOptionsSetup**
https://github.com/Azure/azure-functions-host/blob/04d572f73317597f2b2c28935dae5dcba7f2288d/src/WebJobs.Script/Config/ScriptHostOptionsSetup.cs#L92-L96

- **ExtensionBundleConfigurationHelper**
https://github.com/Azure/azure-functions-host/blob/04d572f73317597f2b2c28935dae5dcba7f2288d/src/WebJobs.Script/Config/ExtensionBundleConfigurationHelper.cs#L48-L51